cicecore/cicedyn/general/ice_init.F90

```diff
@@ -1753,6 +1765,65 @@ subroutine input_data
          abort_list = trim(abort_list)//":29"
       endif
 
+#ifdef USE_PIO1
+      if (history_deflate/=0 .or. restart_deflate/=0 &
+         .or. history_chunksize(1)/=0 .or. history_chunksize(2)/=0 &
+         .or. restart_chunksize(1)/=0 .or. restart_chunksize(2)/=0) then
+         if (my_task == master_task) then
+            write (nu_diag,*) subname//' WARNING: _deflate and _chunksize not compatible with PIO1, will be ignored'
+         endif
+#else
+      if(history_deflate<0 .or. history_deflate>9) then
+         if (my_task == master_task) then
+            write (nu_diag,*) subname//' WARNING: history_deflate value not valid, setting to 0 '
+            write (nu_diag,*) subname//' WARNING: Allowed range: integers from 0 to 9 '
+         endif
+         history_deflate=0
+      endif
+
+      if(restart_deflate<0 .or. restart_deflate>9) then
+         if (my_task == master_task) then
+            write (nu_diag,*) subname//' WARNING: restart_deflate value not valid, setting to 0 '
+            write (nu_diag,*) subname//' WARNING: Allowed range: integers from 0 to 9 '
+         endif
+         restart_deflate=0
+      endif
+
+#ifndef CESMCOUPLED
+      ! history_format not used by nuopc driver
+      if (history_format/='hdf5' .and. history_deflate/=0) then
+         if (my_task == master_task) then
+            write (nu_diag,*) subname//' WARNING: history_deflate not compatible with '//history_format
+            write (nu_diag,*) subname//' WARNING: netcdf compression only possible with history_type="hdf5" '
+         endif
+         history_deflate=0
+      endif
+
+      if (history_format/='hdf5' .and. (history_chunksize(1)/=0 .or. history_chunksize(2)/=0)) then
+         if (my_task == master_task) then
+            write (nu_diag,*) subname//' WARNING: history_chunksize not compatible with '//history_format
+            write (nu_diag,*) subname//' WARNING: netcdf chunking only possible with history_type="hdf5" '
+         endif
+      endif
+
+      if (restart_format/='hdf5' .and. restart_deflate/=0) then
+         if (my_task == master_task) then
+            write (nu_diag,*) subname//' WARNING: restart_deflate not compatible with '//restart_format
+            write (nu_diag,*) subname//' WARNING: netcdf compression only possible with restart_type="hdf5" '
+         endif
+         restart_deflate=0
+      endif
+
+      if (restart_format/='hdf5' .and. (restart_chunksize(1)/=0 .or. restart_chunksize(2)/=0)) then
+         if (my_task == master_task) then
+            write (nu_diag,*) subname//' WARNING: restart_chunksize not compatible with '//restart_format
+            write (nu_diag,*) subname//' WARNING: netcdf chunking only possible with restart_type="hdf5" '
+         endif
+      endif
+
```

Comment:
The advantage is it aborts in `init` and its clear to to the user. 

Otherwise it will ignore the setting (and there would be no warning). 

Yeah you are right - we don't need to set to 0 here.

Reply:
I don't think we should abort here.  The namelist are "unused" in this case, it shouldn't matter what they're set to, don't reset them.  I would say warn (as you have) or do not warn at all.  There are lots of namelist that are set when not used, we generally do not warn users about that.